packages/page-staking/src/Performance/Performance.tsx

```diff
@@ -0,0 +1,321 @@
+// Copyright 2017-2022 @polkadot/app-staking authors & contributors
+// SPDX-License-Identifier: Apache-2.0
+
+import type { SortedTargets } from '../types';
+
+import React, { useCallback, useEffect, useMemo, useState } from 'react';
+
+import { ApiDecoration } from '@polkadot/api/types';
+import { MarkWarning } from '@polkadot/react-components';
+import { useApi, useCall } from '@polkadot/react-hooks';
+import { StorageKey, Vec } from '@polkadot/types';
+import { AccountId, EraIndex, Hash } from '@polkadot/types/interfaces';
+import { AnyTuple, Codec } from '@polkadot/types/types';
+import { Option, Struct, u32 } from '@polkadot/types-codec';
+
+import ActionsBanner from './ActionsBanner';
+import CurrentList from './CurrentList';
+import { SessionEra } from './index';
+import Summary from './Summary';
+
+interface Props {
+  className?: string;
+  favorites: string[];
+  targets: SortedTargets;
+  toggleFavorite: (address: string) => void;
+  sessionEra: SessionEra,
+  currentSessionMode: boolean,
+}
+
+interface CurrentEraValidators extends Struct {
+  reserved: Vec<AccountId>;
+  nonReserved: Vec<AccountId>;
+}
+
+interface CommitteeSize extends Struct {
+  nonReservedSeats: u32
+  reservedSeats: u32,
+}
+
+type SessionIndexEntry = [{ args: [EraIndex] }, Option<u32>];
+
+function Performance ({ className = '', currentSessionMode, favorites, sessionEra, targets, toggleFavorite }: Props): React.ReactElement<Props> {
+  const { api } = useApi();
+
+  const erasStartSessionIndex = useCall<SessionIndexEntry[]>(api.query.staking.erasStartSessionIndex.entries);
+  const [firstBlockInSessionHash, setFirstBlockInSessionHash] = useState<Hash | null>(null);
+  const [lastBlockInSessionHash, setLastBlockInSessionHash] = useState<Hash | null>(null);
+  const [currentEraValidators, setCurrentEraValidators] = useState<CurrentEraValidators | null>(null);
+  const [committeeSize, setCommitteeSize] = useState<CommitteeSize | null>(null);
+  const [sessionValidatorBlockCountLookup, setSessionValidatorBlockCountLookup] = useState<Record<string, number>>({});
+  const [firstSessionBlockAuthor, setFirstSessionBlockAuthor] = useState<string>('');
+  const [isPalletElectionsSupported, setIsPalletElectionsSupported] = useState<boolean>(false);
+
+  const MINIMUM_SUPPORTED_ELECTIONS_PALLET_VERSION = 3;
+
+  const era: number | null = useMemo(() => {
```

Comment:
nit: If possible I would stick to use only one of the null or undefined (preferably undefined)

Reply:
Done